supplementary/alica_ros_proxy/include/clock/AlicaRosTimer.h

```diff
@@ -15,14 +15,13 @@
 namespace alicaRosTimer
 {
 
-template <class CallbackQ>
-class SyncStopTimerRosImpl : public std::enable_shared_from_this<SyncStopTimerRosImpl<CallbackQ>>
+class SyncStopTimerRosImpl : public std::enable_shared_from_this<SyncStopTimerRosImpl>
 {
-    using Base = std::enable_shared_from_this<SyncStopTimerRosImpl<CallbackQ>>;
```

Comment:
you can take the callback queue as a reference parameter from the app as well just like the async spinner. Then you won't need any changes except in the constructors of SyncStopTimerRos/Impl & ThreadPoolRos. Also gives the ability to the app to use different callback queues for alica

Reply:
The app can construct the spinner with the callback queue and pass the spinner alone to the timerfactory, right?

Reply:
but the timer & communication need to put their callbacks into the right callback queue. That's why you see the call to nh.setCallbackQueue in the timer constructor & the ros timer is created from this nh.

Reply:
oh right, i overlooked that. Will fix